src/compiler/checker.ts

```diff
@@ -11409,8 +11409,24 @@ namespace ts {
         }
 
         function checkSourceElement(node: Node): void {
-            if (!node) return;
-            switch (node.kind) {
+            if (!node) {
+                return;
+            }
+
+            let kind = node.kind;

```

Comment:
why not checking timestamp instead, just make sure we check cancellation token if 100 ms passed for instance since last check.